File: 6KYU/Make the Deadfish swim/solution.cs
public class Deadfish
{
  public static int[] Parse(string data)
  {
    var v = 0;
    var outputs = new List<int>();
  
    foreach(var d in data)
    {
      if(d == 'i')
        v++;
      else if(d == 'd')
        v--;
      else if(d == 's')
        v *= v;
      else if(d == 'o')
        outputs.Add(v);
    }
    
    return outputs.ToArray();
  }
}

File: 6KYU/Make the Deadfish swim/solution.go
package kata

func Parse(data string) []int {
	v := 0
	outputs := []int{}

	for i := 0; i < len(data); i++ {
		d := data[i]
		if d == 'i' {
			v = v + 1
		} else if d == 'd' {
			v = v - 1
		} else if d == 's' {
			v *= v
		} else if d == 'o' {
			outputs = append(outputs, v)
		}
	}
	return outputs
}
